### ISSUE TITLE
Does Gaia.launch_job_async work with upload_resource?

### ISSUE THREAD

**Issue by smoh** [astropy/astroquery#963]
- Issue: `Gaia.launch_job_async` with table upload hangs forever. The job page from the url printed indicates the query is PENDING, and this status never changes.

- Code to reproduce the problem:
```python
import tempfile
from astroquery.gaia import Gaia
from astropy.table import Table

f = tempfile.NamedTemporaryFile()
t = Table({'source_id':star['tgas_source_id'].values[:5]})   # just 5 rows of source ids
t.write(f, format='votable')

j = Gaia.launch_job_async(
    "select top 5 * from tap_upload.tmp",
    upload_resource=f.name, upload_table_name='tmp', verbose=True)
```

- The same query with `launch_job` (sync) works.
- `launch_job_async` without upload seems to work fine.
- Full traceback after interrupt [here](https://gist.github.com/smoh/333815c263ef78a37f4add528606ede2)

**Comment by bsipocz** [comment]
cc @jcsegovia

**Comment by jcsegovia** [comment]
I am trying to find the problem.
It seems the job is launched in 'PENDING' state (UWS specification) at server side.
I am sending a 'PHASE=RUN' argument, which, in theory, should inform the server to run the job.
But for some unknown reason, the job never runs.
I need to debug at server side what is going on with that parameter.

**Comment by jcsegovia** [comment]
@bsipocz you have found a bug in Gaia TAP server.
TAP/UWS PHASE parameter is ignored for asynchronous jobs working with an on-the-fly loaded table. 
We are going to include a fix as soon as possible.

**Comment by bsipocz** [comment]
thanks @jcsegovia!

**Comment by StuartLittlefair** [comment]
@bsipocz @jcsegovia Was this ever fixed?

**Comment by jcsegovia** [comment]
It is fixed. It will be available as of Gaia Archive version 2.2.x (DR2)

**Comment by bsipocz** [comment]
Thanks @jcsegovia for confirming.
